src/txn/txn.c

```diff
@@ -1039,8 +1039,48 @@ __txn_fixup_prepared_update(
         hs_cursor->set_value(hs_cursor, &tw, tw.durable_stop_ts, tw.durable_start_ts,
           (uint64_t)WT_UPDATE_STANDARD, &hs_value);
         WT_ERR(hs_cursor->update(hs_cursor));
-    } else
-        WT_ERR(hs_cursor->remove(hs_cursor));
+    } else {
+        /*
+         * Remove the history store entry if checkpoint is not running. Otherwise, fix the history
```

Comment:
I think in general comments could be improved here, I'll rewrite them how I think they should be worded and you can choose to go with them or not.
```
Remove the history store entry if a checkpoint is running, otherwise place a tombstone in front of the history store entry if if it doesn't have a stop timestamp.
```

Reply:
Used the above sentence and modified it a bit as `checkpoint is not running`.